shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -79,16 +91,78 @@ private void init() {
       case surface:
         Log.d(TAG, "Internally creating a FlutterSurfaceView.");
         FlutterSurfaceView flutterSurfaceView = new FlutterSurfaceView(getContext());
+        renderSurface = flutterSurfaceView;
         addView(flutterSurfaceView);
         break;
       case texture:
         Log.d(TAG, "Internally creating a FlutterTextureView.");
         FlutterTextureView flutterTextureView = new FlutterTextureView(getContext());
+        renderSurface = flutterTextureView;
         addView(flutterTextureView);
         break;
     }
   }
 
+  /**
+   * Connects this {@code FlutterView} to the given {@link FlutterEngine}.
+   *
+   * Once invoked, the Flutter UI painted by the given {@link FlutterEngine} will be
+   * displayed by this {@code FlutterView}. Additionally, user touch events, accessibility
+   * events, keyboard events, and more will be forwarded from this {@code FlutterView}
+   * to the attached {@link FlutterEngine}.
+   *
+   * See {@link #detachFromFlutterEngine()} for information on how to detach from a
+   * {@link FlutterEngine}.
+   */
+  public void attachToFlutterEngine(@NonNull FlutterEngine flutterEngine) {
+    if (isAttachedToFlutterEngine()) {
+      if (flutterEngine == this.flutterEngine) {
+        // We are already attached to this FlutterEngine
+        return;
+      }
+
+      // Detach from a previous FlutterEngine so we can attach to this new one.
+      detachFromFlutterEngine();
+    }
+
+    this.flutterEngine = flutterEngine;
+
+    // Instruct our FlutterRenderer that we are now its designated RenderSurface.
+    this.flutterEngine.getRenderer().attachToRenderSurface(renderSurface);
+  }
+
+  /**
+   * Disconnects this {@code FlutterView} from a previously attached {@link FlutterEngine}.
+   *
+   * Once invoked, the UI of this {@code FlutterView} will be cleared, and all touch events,
+   * accessibility events, keyboard events, etc. will no longer be forwarded to the previously
+   * attached {@link FlutterEngine}.
+   *
+   * See {@link #attachToFlutterEngine(FlutterEngine)} for information on how to attach a
+   * {@link FlutterEngine}.
+   */
+  public void detachFromFlutterEngine() {
+    if (!isAttachedToFlutterEngine()) {
+      return;
+    }
+    Log.d(TAG, "Detaching from Flutter Engine");
+
+    // Instruct our FlutterRenderer that we are no longer interested in being its RenderSurface.
+    flutterEngine.getRenderer().detachFromRenderSurface();
+    flutterEngine = null;
+
+    // TODO(mattcarroll): clear the surface when JNI doesn't blow up
+//    if (isSurfaceAvailableForRendering) {
```

Comment:
I suggest creating a branch that contains this code but un-commented.   That way you don't need to check in commented-out code here and watch it bitrot.

Reply:
Is there a benefit to doing that?  Any such branch will rapidly become in conflict with this one, and I'm likely to forget that branch exists and then forget to solve the problem...

Reply:
I'm going to merge as-is to continue the work.  Let's discuss this more if you think it's important to remove it.